pkg/installmanager/installmanager.go

```diff
@@ -1874,6 +1882,35 @@ func isDirNonEmpty(dir string) bool {
 	return err == nil
 }
 
+// pasteInProviderCredentials Add the credentials from a given secret into the ic if nor present
+func pasteInProviderCredentials(icData []byte, cd *hivev1.ClusterDeployment) ([]byte, error) {
+	switch {
+	case cd.Spec.Platform.Nutanix != nil:
+		ic := installertypes.InstallConfig{}
+		if err := yaml.Unmarshal(icData, &ic); err != nil {
+			return nil, errors.Wrap(err, "could not unmarshal InstallConfig")
+		}
+
+		nutanixUsername := os.Getenv(constants.NutanixUsernameEnvVar)
+		nutanixPassword := os.Getenv(constants.NutanixPasswordEnvVar)
+		if nutanixUsername == "" || nutanixPassword == "" {
+			return nil, fmt.Errorf("invalid credentials for %s platform provider", constants.PlatformNutanix)
+		}
+		if ic.Platform.Nutanix.PrismCentral.Username != "" && ic.Platform.Nutanix.PrismCentral.Username != nutanixUsername {
+			return nil, fmt.Errorf("prism central username mismatch for %s platform provider", constants.PlatformNutanix)
+		}
+		if ic.Platform.Nutanix.PrismCentral.Password != "" && ic.Platform.Nutanix.PrismCentral.Password != nutanixPassword {
+			return nil, fmt.Errorf("prism central password mismatch for %s platform provider", constants.PlatformNutanix)
+		}
+
+		ic.Platform.Nutanix.PrismCentral.Username = nutanixUsername
+		ic.Platform.Nutanix.PrismCentral.Password = nutanixPassword
+		return yaml.Marshal(ic)
+	}
+
+	return nil, nil
```

Comment:
```suggestion
	return icData, nil
```
...or you'll be feeding the other platforms an empty install-config.

I believe this is the cause of the e2e[-pool] failures we're seeing in CI.

Reply:
Nice catch, thanks